src/plots/plots.js

```diff
@@ -1051,49 +1059,63 @@ plots.supplyFrameDefaults = function(frameIn) {
     var colorway = layout.colorway || Color.defaults;
     var traceOut = {},
-        defaultColor = colorway[traceOutIndex % colorway.length];
+        defaultColor = colorway[colorIndex % colorway.length];
+
+    var i;
 
     function coerce(attr, dflt) {
         return Lib.coerce(traceIn, traceOut, plots.attributes, attr, dflt);
     }
 
-    function coerceSubplotAttr(subplotType, subplotAttr) {
-        if(!plots.traceIs(traceOut, subplotType)) return;
-
-        return Lib.coerce(traceIn, traceOut,
-            plots.subplotsRegistry[subplotType].attributes, subplotAttr);
-    }
-
     var visible = coerce('visible');
 
     coerce('type');
     coerce('uid');
     coerce('name', layout._traceWord + ' ' + traceInIndex);
 
-    // coerce subplot attributes of all registered subplot types
-    var subplotTypes = Object.keys(subplotsRegistry);
-    for(var i = 0; i < subplotTypes.length; i++) {
-        var subplotType = subplotTypes[i];
-
-        // done below (only when visible is true)
-        // TODO unified this pattern
-        if(['cartesian', 'gl2d'].indexOf(subplotType) !== -1) continue;
-
-        var attr = subplotsRegistry[subplotType].attr;
+    // we want even invisible traces to make their would-be subplots visible
+    // so coerce the subplot id(s) now no matter what
+    var _module = plots.getModule(traceOut);
+    traceOut._module = _module;
+    if(_module) {
+        var basePlotModule = _module.basePlotModule;
+        var subplotAttr = basePlotModule.attr;
+        if(subplotAttr) {
+            var subplots = layout._subplots;
+            var subplotAttrs = basePlotModule.attributes;
+            var subplotId = '';
+
+            // TODO - currently if we draw an empty gl2d subplot, it draws
+            // nothing then gets stuck and you can't get it back without newPlot
+            // sort this out in the regl refactor? but for now just drop empty gl2d subplots
+            if(basePlotModule.name !== 'gl2d' || visible) {
+                if(Array.isArray(subplotAttr)) {
+                    for(i = 0; i < subplotAttr.length; i++) {
+                        var attri = subplotAttr[i];
+                        var vali = Lib.coerce(traceIn, traceOut, subplotAttrs, attri);
+
+                        if(subplots[attri]) Lib.pushUnique(subplots[attri], vali);
+                        subplotId += vali;
+                    }
+                }
+                else {
+                    subplotId = Lib.coerce(traceIn, traceOut, subplotAttrs, subplotAttr);
+                }
 
-        if(attr) coerceSubplotAttr(subplotType, attr);
+                if(subplots[basePlotModule.name]) {
+                    Lib.pushUnique(subplots[basePlotModule.name], subplotId);
+                }
+            }
+        }
     }
 
     if(visible) {
         coerce('customdata');
         coerce('ids');
 
-        var _module = plots.getModule(traceOut);
-        traceOut._module = _module;
```

Comment:
This will conflict with https://github.com/plotly/plotly.js/pull/1869/commits/a02ed20026a80c530f3f40d66dd079ec024bba96

Reply:
I'd be surprised if that's the only conflict this PR makes with regl-2d 😈